### PR TITLE
added write functionality within run with optional flag, defaulting t…

### PR DESCRIPTION
**Pull request by surya-narayanan**
added write functionality within run with optional flag, defaulting to False

### CODE REVIEW

File: pathml/core/slide_data.py [hunk 1]
```diff
@@ -251,6 +251,8 @@ def run(
         level=0,
         tile_pad=False,
         overwrite_existing_tiles=False,
+        write = False,
```
Comment (ryanccarelli) [inline pathml/core/slide_data.py:254]: Could condense these into one argument. If isinstance(write, str or pathlike), write to the path

Comment (jacobrosenthaldfci) [reply pathml/core/slide_data.py]: Agreed

**Comment by jacobrosenthaldfci** [comment]
Thanks for leading this, Surya!!

I modified the syntax a little bit to make it more streamlined (one argument instead of two) and to ensure compatibility with both `SlideData.run()` and also `SlideDataset.run()`. That's why we now pass a directory, instead of a file path. I also added tests.

Let me know what you think! Thanks for suggesting this feature!!

**Comment by surya-narayanan** [comment]
Awesome stuff guys! Let's go!!!

**Review by ryanccarelli — APPROVED** [review]

**Pull request merged**